scripts/variantstore/wdl/HailFromWdl.wdl

```diff
@@ -0,0 +1,250 @@
+                worker_machine_type = worker_machine_type,
+                submission_script = submission_script,
+                variants_docker = GetToolVersions.variants_docker,
+                region = region,
+        }
+    }
+
+    output {
+        Array[File] vcfs = filter_vds_and_export_as_vcf.vcf
+    }
+}
+
+task filter_vds_and_export_as_vcf {
+    input {
+        # You must treat a VDS as a String, since it is a directory and not a single file
+        String vds_url
+        String bed_url
+        String vcf_header_url
+
+        String? git_branch_or_tag
+        File? submission_script
+        String? hail_version
+        String? worker_machine_type
+
+        # contig must be in the reference
+        String contig
+        String prefix
+        String gcs_project
+        String region = "us-central1"
+        Int num_workers
+        RuntimeAttr? runtime_attr_override
+        String gcs_subnetwork_name
+
+        String variants_docker
+    }
+
+    RuntimeAttr runtime_default = object {
+                                      mem_gb: 30,
+                                      disk_gb: 100,
+                                      cpu_cores: 1,
+                                      preemptible_tries: 0,
+                                      max_retries: 0,
+                                      boot_disk_gb: 10
+                                  }
+    RuntimeAttr runtime_override = select_first([runtime_attr_override, runtime_default])
+
+    String default_script_filename = "filter_VDS_and_shard_by_contig.py"
+
+    command <<<
+        # Prepend date, time and pwd to xtrace log entries.
+        PS4='\D{+%F %T} \w $ '
+        set -o errexit -o nounset -o pipefail -o xtrace
+
+        account_name=$(gcloud config list account --format "value(core.account)")
+
+        pip3 install --upgrade pip
+        pip3 install hail~{'==' + hail_version}
+        pip3 install --upgrade google-cloud-dataproc
+
+        if [[ -z "~{git_branch_or_tag}" && -z "~{submission_script}" ]] || [[ ! -z "~{git_branch_or_tag}" && ! -z "~{submission_script}" ]]
+        then
+            echo "Must specify git_branch_or_tag XOR submission_script"
+            exit 1
+        elif [[ ! -z "~{git_branch_or_tag}" ]]
+        then
+            script_url="https://raw.githubusercontent.com/broadinstitute/gatk/~{git_branch_or_tag}/scripts/variantstore/wdl/extract/~{default_script_filename}"
+            curl --silent --location --remote-name "${script_url}"
+        fi
+
+        if [[ ! -z "~{submission_script}" ]]
+        then
+            script_path="~{submission_script}"
+        else
+            script_path="~{default_script_filename}"
+        fi
+
+        # Generate a UUIDish random hex string of <8 hex chars (4 bytes)>-<4 hex chars (2 bytes)>
+        hex="$(head -c4 < /dev/urandom | xxd -p)-$(head -c2 < /dev/urandom | xxd -p)"
+
+        cluster_name="~{prefix}-~{contig}-hail-${hex}"
+        echo ${cluster_name} > cluster_name.txt
```

Comment:
As I understand it, this is the key workaround to make sure we get the cluster name?  We generate the name in advance ourselves, then write it to a file before we even try to start it.  That way, if there's an error anywhere after that the file exists and we at least know the name of the cluster we were *going* to create?

And because we create the file and it's listed as an output, it'll get delocalized even if the workflow is unceremoniously booted from its VM right after that point?

Reply:
Yes that's right, and even if there's a sudden preemption or something the cluster name will be in the logs.